### PR TITLE
Move out event-collector, and reorganizing

### PR DESCRIPTION
**Pull request by jeqo**

### CODE REVIEW

**Comment by jeqo** [comment]
@NikitaZhevnitskiy I have disabled tests, for some reason json files are not recognized. We can fix it after merging.

File: build.sbt [hunk 1]
```diff
@@ -33,35 +33,23 @@ lazy val settings = Seq(
 
                                     /** projects */
 lazy val root = project
-  .in(file("."))
```
Comment (zhenik) [inline build.sbt]: I think we need to have root to be able run `sbt tasks` on root and all aggregated projects. [Ref](https://www.scala-sbt.org/0.13/docs/Multi-Project.html#Aggregation)

File: topic/src/test/scala/no/sysco/middleware/kafka/metadata/topic/rest/TopicMetadataJsonProtocolSpec.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class TopicMetadataJsonProtocolSpec extends WordSpec with Matchers with TopicMetadataJsonProtocol {
+
+  import spray.json._
+
+  //  "TopicVendorProtocol in use, " when {
+  //
+  //    "json => pojo. " should {
+  //
+  //      "Team return proper json " in {
+  //        val json = Utils.jsonFromFile("src/test/resources/team.json")
```
Comment (zhenik) [inline topic/src/test/scala/no/sysco/middleware/kafka/metadata/topic/rest/TopicMetadataJsonProtocolSpec.scala]: Use project root `topic/src/test/resources/team.json`

File: topic/src/test/scala/no/sysco/middleware/kafka/metadata/topic/rest/TopicMetadataJsonProtocolSpec.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  //            | {
+  //            |   "name": "yes"
+  //            | }
+  //          """.stripMargin
+  //        val thrown = intercept[DeserializationException] { JsonParser(json).convertTo[Team] }
+  //        assert(thrown.getMessage === "Object is missing required member 'department'")
+  //      }
+  //
+  //      "TopicMetadata return proper pojo " in {
+  //        val json = Utils.jsonFromFile("src/test/resources/topic-metadata.json")
```
Comment (zhenik) [inline topic/src/test/scala/no/sysco/middleware/kafka/metadata/topic/rest/TopicMetadataJsonProtocolSpec.scala]: Use project root `topic/src/test/resources/topic-metadata.json`

File: topic/src/main/scala/no/sysco/middleware/kafka/metadata/topic/AppConfig.scala [hunk 4]
```diff
@@ -14,7 +14,7 @@ case class ApplicationConfig(kafka: KafkaConfig, rest: RestConfig, env: Env.Valu
 object AppConfig {
 
   def loadConfig(): ApplicationConfig = {
-    val config = ConfigFactory.load("application.conf")
+    val config = ConfigFactory.load("src/main/resources/application.conf")
```
Comment (zhenik) [inline topic/src/main/scala/no/sysco/middleware/kafka/metadata/topic/AppConfig.scala]: Change to 
```
val config = ConfigFactory.load()
```

File: build.sbt (hunk 1 above)
Comment (zhenik) [reply build.sbt]: It should be smth like [here](https://github.com/NikitaZhevnitskiy/sbt-multimodule-example/blob/master/build.sbt) or [here kafka-security](https://github.com/simplesteph/kafka-security-manager/blob/master/build.sbt#L9)

File: build.sbt [hunk 5]
```diff
@@ -33,6 +33,21 @@ lazy val settings = Seq(
 
                                     /** projects */
 lazy val root = project
+  .in(file("."))
+  .settings(
+    name := "kafka-metadata",
+    organization := "no.sysco.middleware.kafka.metadata",
+    version := projectV,
+    settings,
+    libraryDependencies ++= commonDependencies ++ observabilityDependencies ++ testDependencies,
```
Comment (zhenik) [inline build.sbt:42]: I root does not need include 
```
    libraryDependencies ++= commonDependencies ++ observabilityDependencies ++ testDependencies,
    mainClass in assembly := Some(s"no.sysco.middleware.kafka.metadata.topic.Main"),
    assemblyJarName in assembly := "ktm-fat-jar.jar"
```

Comment (jeqo) [reply build.sbt:42]: I think we can fix this after merge.

Comment (zhenik) [reply build.sbt:42]: aggree

**Comment by zhenik** [comment]
Could u please resolve conflicts and merge this PR. I fix rest later

**Pull request merged**